File: packages/plugin-simplicity/fixtures/go/archive.go
package fixtures

import (
	"errors"
	"io/fs"
	"os"
)

func ReadArchive(path string) []byte {
	bytes, err := os.ReadFile(path)
	if isMissing(err) {
		return nil
	}
	return bytes
}

// An exact match: copied character for character from store.go.
func isMissing(err error) bool {
	return errors.Is(err, fs.ErrNotExist)
}

File: packages/plugin-simplicity/fixtures/rust/archive.rs
use std::io;

pub fn read_archive(path: &str) -> Vec<u8> {
    match std::fs::read(path) {
        Ok(bytes) => bytes,
        Err(error) if is_missing(&error) => Vec::new(),
        Err(error) => panic!("{}", error),
    }
}

// An exact match: copied character for character from `store.rs`.
fn is_missing(error: &io::Error) -> bool {
    error.kind() == io::ErrorKind::NotFound
}
